src/main/java/edu/wpi/first/shuffleboard/WidgetPaneController.java

```diff
@@ -95,6 +109,54 @@ private void initialize() {
       cleanupWidgetDrag();
       event.consume();
     });
+
+    pane.parentProperty().addListener((__, old, parent) -> {
+      if (parent instanceof Region) {
+        Region region = (Region) parent;
+        Binding<Integer> colBinding =
+            EasyBind.combine(region.widthProperty(), pane.hgapProperty(), pane.tileSizeProperty(),
+                (width, gap, size) -> pane.roundWidthToNearestTile(width.doubleValue(), RoundingMode.DOWN))
+                .map(numCols -> Math.max(1, numCols));
+        Binding<Integer> rowBinding =
+            EasyBind.combine(region.heightProperty(), pane.vgapProperty(), pane.tileSizeProperty(),
+                (height, gap, size) -> pane.roundHeightToNearestTile(height.doubleValue(), RoundingMode.DOWN))
+                .map(numRows -> Math.max(numRows, 1));
+
+
+        pane.numColumnsProperty().bind(colBinding);
+        pane.numRowsProperty().bind(rowBinding);
+      }
+    });
+
+    pane.numColumnsProperty().addListener((__, oldCount, newCount) -> {
+      if (pane.getTiles().isEmpty()) {
+        // No tiles, bail
+        return;
+      }
+      if (newCount < oldCount) {
+        // shift and shrink tiles to the left
+        pane.getTiles().stream()
+            .filter(tile -> {
+              final TileLayout layout = pane.getTileLayout(tile);
+              return layout.origin.col + layout.size.getWidth() > newCount;
+            })
+            .forEach(tile -> moveTileLeft(tile, oldCount - newCount));
```

Comment:
This will move the first tile left N times, then the second tile, etc. I think it's probably better to move EVERY tile once, then every tile again, etc. Basically, maybe move the `for (int i = 0; i < count; i++)` to this function instead of the moveTileLeft.